apps/user_ldap/lib/Migration/Version1141Date20220323143801.php

```diff
@@ -0,0 +1,98 @@
+<?php
+
+declare(strict_types=1);
+
+namespace OCA\User_LDAP\Migration;
+
+use Closure;
+use OCP\DB\ISchemaWrapper;
+use OCP\DB\QueryBuilder\IQueryBuilder;
+use OCP\IDBConnection;
+use OCP\Migration\IOutput;
+use OCP\Migration\SimpleMigrationStep;
+
+/**
+ * Auto-generated migration step: Please modify to your needs!
+ */
+class Version1141Date20220323143801 extends SimpleMigrationStep {
+
+	private IDBConnection $dbc;
+
+	public function __construct(IDBConnection $dbc) {
+		$this->dbc = $dbc;
+	}
+
+	/**
+	 * @param IOutput $output
+	 * @param Closure $schemaClosure The `\Closure` returns a `ISchemaWrapper`
+	 * @param array $options
+	 */
+	public function preSchemaChange(IOutput $output, Closure $schemaClosure, array $options): void {
+		foreach (['ldap_user_mapping', 'ldap_group_mapping'] as $tableName) {
+			$dnsTooLong = [];
+			$lengthExpr = $this->dbc->getDatabasePlatform()->getLengthExpression('ldap_dn');
+
+			$qb = $this->dbc->getQueryBuilder();
+			$qb->select('ldap_dn')
+				->from($tableName)
+				->where($qb->expr()->gt($qb->createFunction($lengthExpr), '255', IQueryBuilder::PARAM_INT));
+
+			$result = $qb->executeQuery();
+			while($dn = $result->fetchOne()) {
```

Comment:
```suggestion
			while(($dn = $result->fetchOne()) !== false) {
```
Let’s be extra cautious with those now, we had trouble in the past.